target/cheri-common/op_helper_cheri_common.c

```diff
@@ -703,6 +775,11 @@ void CHERI_HELPER_IMPL(cfromptr(CPUArchState *env, uint32_t cd, uint32_t cb,
 #ifdef DO_CHERI_STATISTICS
     OOB_INFO(cfromptr)->num_uses++;
 #endif
+
+    bool not_null = (cb & CFROMPTR_0_IS_NOT_NULL) != 0;
```

Comment:
This has changed a little since you last looked at it. cheri_jump_and_link is now the factored out version with a flags argument (as you suggest). I am still passing flags through one of the register arguments in CJALR because I didn't want to duplicate it. Should I just duplicate it with an extra flags argument?